src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,35 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    public static final int MIN_AGE = 18;
+    public static final int MIN_PASSWORD_LENGTH = 6;
+    public static final int MIN_LOGIN_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login does not by null");
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password does not by null");
+        }
+        if (user.getAge() == null) {
+            throw new RegistrationException("Age does not be null");
+        }
+        if (user.getAge() < MIN_AGE) {
+            throw new RegistrationException("Age does not be null" + MIN_AGE);
+        }
+        if (user.getLogin().length() < MIN_LOGIN_LENGTH) {
+            throw new RegistrationException("Are no enough characters" + MIN_LOGIN_LENGTH);
+        }
+        if (user.getPassword().length() < MIN_PASSWORD_LENGTH) {
+            throw new RegistrationException("Are no enough characters" + MIN_PASSWORD_LENGTH);
```

Comment:
learn how to write good error messages
```suggestion
            throw new RegistrationException("Login can't be null");
        }
        if (user.getPassword() == null) {
            throw new RegistrationException("Password can't be null");
        }
        if (user.getAge() == null) {
            throw new RegistrationException("Age can't be null");
        }
        if (user.getAge() < MIN_AGE) {
            throw new RegistrationException("INVALID AT ALL, REWRITE THIS MESSAGE" + MIN_AGE);
        }
        if (user.getLogin().length() < MIN_LOGIN_LENGTH) {
            throw new RegistrationException("Login should be at least " + MIN_LOGIN_LENGTH + " characters long");
        }
        if (user.getPassword().length() < MIN_PASSWORD_LENGTH) {
            throw new RegistrationException("REWRITE ME AS IN PREV EXAMPLE" + MIN_PASSWORD_LENGTH);
```